test/e2e/specs/editor/blocks/comments.spec.js

```diff
@@ -141,35 +144,210 @@ test.describe( 'Comments', () => {
 		await requestUtils.deleteAllComments();
 	} );
 
 	test.afterAll( async ( { requestUtils } ) => {
 		await requestUtils.activateTheme( 'twentytwentyone' );
 	} );
+
+	test( 'is still supported', async ( { page, requestUtils } ) => {
+		// Create a post with the old "Post Comments" block.
+		const { id: postId } = await requestUtils.createPost( {
+			content: '<!-- wp:post-comments /-->',
+			status: 'publish',
+		} );
+
+		// Publish a comment on that post.
+		await requestUtils.createComment( {
+			content: 'This is an automated comment',
+			post: postId,
+		} );
+
+		// Visit created post.
+		await page.goto( `/?p=${ postId }` );
+
+		// Ensure that the rendered post is the legacy version of Post Comments.
+		await expect( page.locator( '.wp-block-post-comments' ) ).toBeVisible();
+		await expect( page.locator( '.comment-content' ) ).toContainText(
+			'This is an automated comment'
+		);
+	} );
+
+	test( 'is converted to Comments with legacy attribute', async ( {
+		page,
+		admin,
+		requestUtils,
+		codeEditorUtils,
+		commentsBlockUtils,
+	} ) => {
+		// Create a post with the old "Post Comments" block.
+		const { id: postId } = await requestUtils.createPost( {
+			content: '<!-- wp:post-comments /-->',
+			status: 'publish',
+		} );
+		await requestUtils.createComment( {
+			content: 'This is an automated comment',
+			post: postId,
+		} );
+
+		// Go to the post editor.
+		await admin.visitAdminPage(
+			'/post.php',
+			`post=${ postId }&action=edit`
+		);
+
+		// Hide welcome guide.
+		await commentsBlockUtils.hideWelcomeGuide();
+
+		// Check that the Post Comments block has been replaced by Comments (legacy)
+		await expect( page.locator( '.wp-block-post-comments' ) ).toBeHidden();
+		await expect( page.locator( '.wp-block-comments' ) ).toBeVisible();
+
+		// Open code editor and check its content.
+		await codeEditorUtils.open();
+		await expect( codeEditorUtils.textbox() ).toContainText(
+			'<!-- wp:comments {"legacy":true} /-->'
+		);
+		await codeEditorUtils.close();
+
+		// Visit post
+		await page.goto( `/?p=${ postId }` );
+
+		// Rendered block should be the same as Post Comments
+		await expect( page.locator( '.wp-block-post-comments' ) ).toBeVisible();
+		await expect( page.locator( '.comment-content' ) ).toContainText(
+			'This is an automated comment'
+		);
+	} );
 } );
 
 class CommentsBlockUtils {
-	constructor( { page, admin } ) {
+	constructor( { page, admin, requestUtils } ) {
 		this.page = page;
 		this.admin = admin;
+		this.requestUtils = requestUtils;
+		this.previousOptions = {};
+	}
+
+	async setOptions( options ) {
+		for ( const [ key, value ] of Object.entries( options ) ) {
+			const previousValue = await this.setOption( key, value );
+			if ( ! this.previousOptions[ key ] ) {
+				this.previousOptions[ key ] = previousValue;
+			}
+		}
+	}
+
+	async restorePreviousOptions() {
+		for ( const [ key, value ] of Object.entries( this.previousOptions ) ) {
+			await this.setOption( key, value );
+		}
```

Comment:
I normally like to avoid statefulness in tools like this, but it looks quite handy here 🤔 

What gives me a bit of pause is that `setOption` continues to be stateless (and returns the old value), whereas `setOptions` works quite differently (stores the previous values internally, doesn't return anything). That's a bit inconsistent and potentially confusing.

As an alternative, we could considering a stateless `setOptions` that returns an object of the same shape as its argument -- keys are settings, values are the previous settings' values.

(Or is the stateful approach maybe encouraged by Playwright?)

Reply:
> (Or is the stateful approach maybe encouraged by Playwright?)

I wouldn't say it is encouraged, although the fixtures API is designed so you can control setup and teardown for tests without using having to use `test.beforeEach` or `test.afterEach`. So, it could make sense in this case? 🤔

There are more examples in the [docs](https://playwright.dev/docs/test-fixtures).

In any case, I understand your concerns about the different behavior between `setOption` and `setOptions`, and I think it would be better if we make the latter _stateless_ as you suggested, just for the sake of consistency.

Reply:
I've made `setOptions` stateless. See https://github.com/WordPress/gutenberg/pull/41807/commits/cf790fde980858ca59255385d6544b1ae41349dc.